test/sequential/test-fs-readdir-recursive.js

```diff
@@ -79,99 +79,349 @@ function createFiles(path, fileStructure) {
 // Make sure tmp directory is clean
 tmpdir.refresh();
 
+const results = [
+  [
+    'a', 'a/bar', 'a/foo', 'aa', 'aa/bar', 'aa/foo',
```

Comment:
```suggestion
    'a', 'a/a', 'a/bar', 'a/foo', 'aa', 'aa/bar', 'aa/foo',
```
There are test failures, we should expect `'a/a'` directory in outputs where it's not filtered out.